kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+                        null
+                    )
+                } catch (e: Exception) {
+                    Toast.makeText(
+                        this,
+                        "Error populating OGC Feature Collection Table from service: " +
+                                e.message,
+                        Toast.LENGTH_LONG
+                    ).show()
+                    Log.e(
+                        "OGC Service error: ",
+                        e.message.toString()
+                    )
+                }
+            } else {
+                Toast.makeText(
+                    this,
+                    "Failed to load OGC Feature Collection Table" +
+                            ogcFeatureCollectionTable.loadError.message,
+                    Toast.LENGTH_LONG
+                ).show()
+                Log.e(
+                    "OGC Load error: ",
+                    ogcFeatureCollectionTable.loadError.message.toString()
+                )
+            }
+        }
+
+        // Load the table
+        ogcFeatureCollectionTable.loadAsync()
+
+        // Handles CQL Filters in a Bottom Sheet View.
+        fab.setOnClickListener {
+            openBottomSheetView()
+        }
+    }
+
+    /**
+     * Handles CQL Filters in a Bottom Sheet View.
+     */
+    private fun openBottomSheetView() {
+
+        // Resets views in bottomSheet to default values.
+        resetBottomSheetValues()
+
+        // Creates a new BottomSheetDialog
+        val dialog = BottomSheetDialog(this)
+
+        // Inflates layout file
+        val cqlFiltersLayoutBinding = CqlFiltersLayoutBinding.inflate(layoutInflater)
+
+        // Sets the Where Clause for CQL filter
+        cqlFiltersLayoutBinding.whereClauseLayout.setOnClickListener {
+
+            // Creates a dialog to choose a where clause
+            val alertDialog: AlertDialog.Builder = AlertDialog.Builder(this)
+            alertDialog.setTitle("AlertDialog")
+            val checkedItem = cqlQueryListPosition
+
+            alertDialog.setSingleChoiceItems(
+                cqlQueryList.toTypedArray(), checkedItem
+            ) { dialog, which ->
+
+                // Updates the selected where clause
+                cqlQueryListPosition = which
+                cqlFiltersLayoutBinding.cqlQueryTextView.text = cqlQueryList[which]
+
+                // Dismiss dialog
+                dialog.dismiss()
+            }
+
+            // Displays the where clause dialog
+            val alert: AlertDialog = alertDialog.create()
+            alert.show()
+        }
+
+        // Sets the view to the default value of max features (1000)
+        cqlFiltersLayoutBinding.maxFeaturesEditText.setText(maxFeatures.toString())
+
+        // Sets from date to Jun-13-2011 by default
+        cqlFiltersLayoutBinding.fromDatePicker.updateDate(2011, 5, 13)
+        // Sets to date to Jan-7-2012 by default
+        cqlFiltersLayoutBinding.toDatePicker.updateDate(2012, 0, 7)
```

Comment:
today I learned Android uses index 0 to refer to the first month of the year!

Reply:
I didn't know about this either!